### PR TITLE
ID-36: Check if passwords are comprimised using Have I Been Pwned

### PR DESCRIPTION
**Pull request by bdsl**

### CODE REVIEW

File: src/Application/Actions/Person/Update.php [hunk 1]
```diff
@@ -129,7 +129,7 @@ protected function action(Request $request, array $args): Response
         $violations = $this->validator->validate($person, null, ['complete']);
 
         if (count($violations) > 0) {
-            $message = 'Validation error: ';
+            $message = '';
```
Comment (bdsl) [inline src/Application/Actions/Person/Update.php:132]: Removed this to make the message friendlier to display on the front end site.

File: composer.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
         "stripe/stripe-php": "^9.1.0",
         "symfony/cache": "^6.1",
         "symfony/doctrine-bridge": "^6.1",
+        "symfony/http-client": "^6.2",
```
Comment (NoelLH) [inline composer.json:35]: Do we benefit from adding this on top of Guzzle? Do they share PSR interfaces these days?

**Review by NoelLH — COMMENTED** [review]
Looking good, just want to clarify the HTTP client dependency choice

File: composer.json (hunk 2 above)
Comment (bdsl) [reply composer.json:35]: No, they don't share an interface - or if they do Symfony also implements its own interface. The Symfony NotComprimised validation rule doesn't work without it - it just triggers an error that tells you to install this. If we avoided this we'd have to use a different way to connect to the API. There's a lot on packagist but none else from anywhere with brand recognition comparable to Symfony. https://packagist.org/?query=haveibeenpowned

**Review by NoelLH — DISMISSED** [review]

**Comment by bdsl** [comment]
(Do not merge without reviewing Jira comments)

**Pull request merged**